src/Main.elm

```diff
@@ -1,10 +1,197 @@
+                    { newOpened
+                        | currentId = Just id
+                        , parentId = newParentId
+                        , parentName = newParentName name
+                        , usedIdList = newUsedIdList id
+                        , state = True
+                    }
+                , breadcrumbs =
+                    { newBreadcrumb
+                        | breadCrumbName = newBreadcrumbpName name
+                        , breadCrumbId = newBreadcrumbId id
+                    }
+            }
+
+        MsgFolderClosed ->
+            { model
+                | opened =
+                    { newOpened
+                        | currentId = newCurrentId
+                        , parentId = List.head (List.drop 2 newOpened.usedIdList)
+                        , parentName = List.drop 1 newOpened.parentName
+                        , usedIdList = List.drop 1 newOpened.usedIdList
+                        , state = stateOfcurrentId newCurrentId
+                    }
+                , breadcrumbs =
+                    { newBreadcrumb
+                        | breadCrumbName = List.drop 1 newBreadcrumb.breadCrumbName
+                        , breadCrumbId = List.drop 1 newBreadcrumb.breadCrumbId
+                    }
+            }
+
+
+view : Model -> Html Msg
+view model =
+    let
+        isOpened =
+            model.opened.state
+    in
+    Html.div [ class "container" ]
+        [ Html.h1 [ class "h1" ]
+            [ Html.text "Audience Browser"
+            ]
+        , Html.div [ class "listContainer" ]
+            [ Html.div [ id "head" ]
+                [ Html.div [ class "breadcrumbsArea" ]
+                    [ Html.a [ class "breadcrumbsText" ] [ text "Home" ]
+                    , Html.ul [ class "breadcrumbsText" ] (List.map (breadCrumbs model.breadcrumbs) (List.reverse model.breadcrumbs.breadCrumbName))
+                    ]
+                , if isOpened == True then
+                    Html.button [ class "backButton", onClick MsgFolderClosed ] [ text "Go Up" ]
+
+                  else
+                    text ""
+                ]
+            , Html.ul [ class "list" ]
+                (List.map (openFolder model.opened) model.folders)
+            , Html.ul [ class "list" ]
+                (List.map (viewAudience model.opened) model.audience)
+            ]
+        ]
+
+
+openFolder : Opened -> AudFolders.AudienceFolder -> Html Msg
+openFolder opened list =
+    let
+        folderButton =
+            Html.button [ class "folder", onClick (MsgFolderOpened list.id list.name) ] [ text list.name ]
+    in
+    if opened.currentId == list.parent && opened.state == True then
+        folderButton
+
+    else if list.parent == Nothing && opened.state == False then
+        folderButton
+
+    else
+        text ""
+
+breadCrumbs : Breadcrumbs -> String -> Html msg
+breadCrumbs breadcrumbs list =
+    li [ class "breadcrumbs" ] [ text (" -> " ++ list) ]
```

Comment:
I know it's not forced in pro-next code and it's controversial topic, but I prefer to keep function definitions before first usage. Usually when you read code it's from top to bottom and when you have function defined first you already know what is this function about later when it's called. But this rule is somehow not relevant for big files as we have in Elm or in case you have tens of function defined before it's used, because you can't hold everything in your head. In cases like that it's important to have good naming. Here I would consider `breadcrumbView` so it's clear purpose of this function.